File: app/src/main/java/com/zillennium/utswap/screens/setting/lockUpScreen/adapter/LockUpAdapter.java
package com.zillennium.utswap.screens.setting.lockUpScreen.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.zillennium.utswap.R;
import com.zillennium.utswap.models.LockUpHistory.LockUpHistory;

import java.util.ArrayList;

public class LockUpAdapter extends RecyclerView.Adapter<LockUpAdapter.ViewHolder> {

    private ArrayList<LockUpHistory> arrayList = new ArrayList<>();
    private LockUpAdapter.OnClickLockUpHistory onClickLockUpHistory;

    public LockUpAdapter(ArrayList<LockUpHistory> arrayList, OnClickLockUpHistory onClickLockUpHistory) {
        this.arrayList = arrayList;
        this.onClickLockUpHistory = onClickLockUpHistory;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list_wallet_lock_up_history,parent,false);
        return new LockUpAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        LockUpHistory lockUpHistory = arrayList.get(position);

        holder.txtAmount.setText(lockUpHistory.getAmount());
        holder.txtDate.setText(lockUpHistory.getDate());

        if (arrayList.size() == 1) {
            holder.line.setVisibility(View.GONE);
        }else {
            if (arrayList.size() - 1 == position){
                holder.line.setVisibility(View.GONE);
            } else if (0 == position) {
                holder.itemView.setBackgroundResource(R.drawable.card_view_whtie_border_top);
                holder.line.setVisibility(View.VISIBLE);
            } else {
                holder.itemView.setBackgroundResource(R.color.white);
                holder.line.setVisibility(View.VISIBLE);
            }
        }

        holder.itemView.setOnClickListener(view -> {
            onClickLockUpHistory.clickMe(lockUpHistory);
        });
    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private TextView txtAmount, txtDate;
        private View line;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            txtAmount = itemView.findViewById(R.id.txt_amount);
            txtDate = itemView.findViewById(R.id.txt_date);
            line = itemView.findViewById(R.id.line);
        }
    }

    public interface OnClickLockUpHistory{
        void clickMe(LockUpHistory lockUpHistory);
    }
}

File: app/src/main/java/com/zillennium/utswap/screens/wallet/lockUpScreen/adapter/LockUpAdapter.kt
package com.zillennium.utswap.screens.wallet.lockUpScreen.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.zillennium.utswap.R
import com.zillennium.utswap.models.LockUpHistory.LockUpHistory


class LockUpAdapter(
    arrayList: ArrayList<LockUpHistory>,
    onClickLockUpHistory: OnClickLockUpHistory
) :
    RecyclerView.Adapter<LockUpAdapter.ViewHolder>() {
    private var arrayList = ArrayList<LockUpHistory>()
    private val onClickLockUpHistory: OnClickLockUpHistory
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_list_wallet_lock_up_history, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val lockUpHistory = arrayList[position]
        holder.txtAmount.text = lockUpHistory.amount
        holder.txtDate.text = lockUpHistory.date
        if (arrayList.size == 1) {
            holder.line.visibility = View.GONE
        } else {
            when (position) {
                arrayList.size - 1 -> {
                    holder.line.visibility = View.GONE
                }
                0 -> {
                    holder.itemView.setBackgroundResource(R.drawable.card_view_whtie_border_top)
                    holder.line.visibility = View.VISIBLE
                }
                else -> {
                    holder.itemView.setBackgroundResource(R.color.white)
                    holder.line.visibility = View.VISIBLE
                }
            }
        }
        holder.itemView.setOnClickListener {
            onClickLockUpHistory.clickMe(
                lockUpHistory
            )
        }
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val txtAmount: TextView = itemView.findViewById(R.id.txt_amount)
        val txtDate: TextView = itemView.findViewById(R.id.txt_date)
        val line: View = itemView.findViewById(R.id.line)

    }

    interface OnClickLockUpHistory {
        fun clickMe(lockUpHistory: LockUpHistory?)
    }

    init {
        this.arrayList = arrayList
        this.onClickLockUpHistory = onClickLockUpHistory
    }
}
